fe/fe-core/src/main/java/com/starrocks/connector/ConnectorPartitionTraits.java

```diff
@@ -483,8 +484,59 @@ PartitionKey createEmptyKey() {
         @Override
         public Set<String> getUpdatedPartitionNames(List<BaseTableInfo> baseTables,
                                                     MaterializedView.AsyncRefreshContext context) {
-            // TODO: implement
-            return null;
+            PaimonTable baseTable = (PaimonTable) table;
+            Set<String> result = Sets.newHashSet();
+            for (BaseTableInfo baseTableInfo : baseTables) {
+                if (!baseTableInfo.getTableIdentifier().equalsIgnoreCase(baseTable.getTableIdentifier())) {
+                    continue;
+                }
+                Map<String, MaterializedView.BasePartitionInfo> partitionVersionMap =
+                        context.getBaseTableRefreshInfo(baseTableInfo);
+                List<String> partitions = getPartitionNames();
```

Comment:
Could there be any impact on performance? If so, please include a note about it, and we can consider optimization later on.